### PR TITLE
support for boto profiles

### PR DESCRIPTION
**Pull request by bootswithdefer**
Support for multiple AWS accounts with credentials specified by profile name.

This also contains a fix for multiple AWS sources.  Placebo doesn't work with multiple boto sessions being stored in the same directory, so this patch creates subdirectories.

I can split into two patches if needed.

### CODE REVIEW

File: bridgy/inventory/__init__.py [hunk 1]
```diff
@@ -22,10 +22,20 @@ def inventory(config):
 
     for source, srcCfg in config.sources():
         if source == 'aws':
-            if os.path.exists(os.path.expanduser("~/.aws")):
-                inventory = AwsInventory(cache_dir=config.inventoryDir(AwsInventory.name))
+            cache_dir = config.inventoryDir(AwsInventory.name)
+            if srcCfg['name'] != None:
+                cache_dir = os.path.join(cache_dir, srcCfg['name'])
```
Comment (wagoodman) [inline bridgy/inventory/__init__.py]: This path joining could be moved to `config.inventoryDir()` here https://github.com/wagoodman/bridgy/blob/master/bridgy/config/base.py#L82 .

File: bridgy/inventory/__init__.py [hunk 2]
```diff
@@ -22,10 +22,20 @@ def inventory(config):
...
+                if not os.path.exists(cache_dir):
```
Comment (wagoodman) [inline bridgy/inventory/__init__.py]: Similarly, path creations have been reserved for the `config.create()` method https://github.com/wagoodman/bridgy/blob/master/bridgy/config/base.py#L56

File: bridgy/version.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-__version__ = '0.1.0'
+__version__ = '0.1.1'
```
Comment (wagoodman) [inline bridgy/version.py]: I accidentally stole this patch number from you already :)

**Comment by wagoodman** [comment]
I just remembered that I'm using the same placebo stub for all aws tests... they were all derived from the same `boto.Session`. You've introduced a new `boto.Session` that has not been recorded by placebo yet. Once you add a stub in this dir https://github.com/bootswithdefer/bridgy/tree/master/tests/aws_stubs I think your tests should start working. (Just a guess, I haven't tried it out yet... probably not till tonight)

**Comment by bootswithdefer** [comment]
I've cleaned up inventoryDir, however config.create doesn't look like it has any of the config loaded when it's called so I can't create subdirs based on inventory names.

The testing issue appears to be that you need a boto config (~/.aws/config or ~/.aws/credentials) with a default section.  Not sure how to have tox the build deal with it.  Maybe we need to tell Travis to create it?

**Comment by wagoodman** [comment]
Yeah, you're right, I forgot that `create()` is called before `read()`, so there isn't a config yet --small but important detail :). I'll play with the failing test tonight... it's really just a matter of getting the test harness right, not really fair to throw in this PR. Since I don't cut new pypi builds automatically from master, I'll merge this for now and tweak the test.

**Comment by wagoodman** [comment]
@bootswithdefer just uploaded v0.1.2 with this feature.

**Pull request merged**